service/src/main/kotlin/fi/espoo/evaka/pis/controllers/EmployeeController.kt

```diff
@@ -149,20 +149,26 @@ class EmployeeController {
         }
     }
 
-    @GetMapping("/search")
+    @PostMapping("/search")
     fun searchEmployees(
         db: Database.Connection,
         user: AuthenticatedUser,
-        @RequestParam(required = false) page: Int?,
-        @RequestParam(required = false) pageSize: Int?,
-        @RequestParam(required = false) searchTerm: String?,
+        @RequestBody body: SearchEmployeeRequest
     ): ResponseEntity<Paged<EmployeeWithDaycareRoles>> {
         Audit.EmployeesRead.log()
         user.requireOneOfRoles(UserRole.ADMIN)
+        val maxPageSize = 5000
+        if (body.pageSize != null && body.pageSize > maxPageSize) throw BadRequest("Maximum page size is $maxPageSize")
         return db.read { tx ->
-            getEmployeesPaged(tx, page ?: 1, (pageSize ?: 50).coerceAtMost(100), searchTerm ?: "")
+            getEmployeesPaged(tx, body.page ?: 1, (body.pageSize ?: 50).coerceAtMost(100), body.searchTerm ?: "")
```

Comment:
Page size is already at most `100` (see `(body.pageSize ?: 50).coerceAtMost(100)`) so the first page size check is kind of conflicting with it.